tests/integration/queries/test_migration_progress.py

```diff
@@ -0,0 +1,353 @@
+        for schema in ("schema1", "schema2")
+        for table in ("table1", "table2", "table3", "table4", "table5")
+    ):
+        record = ("tables", [table.catalog, table.database, table.name], table, [], "Cor")
+        records.append(record)
+    return records
+
+
+@pytest.fixture
+def table_migration_statuses(tables):
+    statuses = []
+    for _, id_, table, _, owner in tables:
+        table_migration_status = TableMigrationStatus(table.catalog, table.database, table.name)
+        failures = ["not migrated"]
+        owner = "Andrew" if table.name == "table1" else "Cor"
+        if table.database == "schema1":  # Simulate one schema being migrated
+            if table.name == "table2":  # An owner with only migrated objects
+                owner = "Eric"
+            table_migration_status.dst_catalog = "catalog1"
+            table_migration_status.dst_schema = table.database
+            table_migration_status.dst_table = table.name
+            failures = []
+        statuses.append(("migration_status", id_, table_migration_status, failures, owner))
+    return statuses
+
+
+@pytest.fixture
+def udfs():
+    records = []
+    for udf in (
+        Udf(
+            "hive_metastore",
+            "schema1",
+            "custom_function",
+            deterministic=True,
+            data_access="UNKNOWN",
+            body="UNKNOWN",
+            comment="UNKNOWN",
+            func_type="UNKNOWN",
+            func_input="UNKNOWN",
+            func_returns="UNKNOWN",
+        ),
+        Udf(
+            "hive_metastore",
+            "schema1",
+            "invalid_function",
+            deterministic=True,
+            data_access="UNKNOWN",
+            body="UNKNOWN",
+            comment="UNKNOWN",
+            failures="UDF not supported by UC",
+            func_type="UNKNOWN",
+            func_input="UNKNOWN",
+            func_returns="UNKNOWN",
+        ),
+    ):
+        record = (
+            "udfs",
+            [udf.catalog, udf.database, udf.name],
+            udf,
+            [] if not udf.failures else udf.failures.split("\n"),
+            "Cor",
+        )
+        records.append(record)
+    return records
+
+
+@pytest.fixture
+def grants():
+    records = []
+    for grant in (
+        Grant("service_principal", "MODIFY", "hive_metastore"),
+        Grant("Eric", "OWN", "hive_metastore", "sales"),
+        Grant("Liran", "DENY", "hive_metastore", "sales"),
+    ):
+        record = (
+            "grants",
+            [grant.principal, grant.action_type],
+            grant,
+            ["DENY is not supported by UC"] if grant.action_type == "DENY" else [],
+            "Cor",
+        )
+        records.append(record)
+    return records
+
+
+@pytest.fixture
+def jobs():
+    records = []
+    for job in (
+        JobInfo("1", success=1, failures=""),
+        JobInfo("2", success=0, failures="No isolation shared clusters not supported in UC"),
```

Comment:
```suggestion
        JobInfo("2", success=0, failures=`["No isolation shared clusters not supported in UC"]`),
```
`failures` is a JSON-serialized array of strings